contracts/axelarnet-gateway/src/contract/execute.rs

```diff
@@ -129,20 +141,48 @@ pub fn call_contract(
 
 pub fn route_messages(
     storage: &mut dyn Storage,
+    querier: QuerierWrapper,
     sender: Addr,
     msgs: Vec<Message>,
 ) -> Result<Response<nexus::execute::Message>> {
     let Config {
-        chain_name, router, ..
+        chain_name,
+        router,
+        nexus,
     } = state::load_config(storage);
+
     let router = Router::new(router);
+    let client: nexus::Client = client::CosmosClient::new(querier).into();
 
-    if sender == router.address {
-        Ok(prepare_msgs_for_execution(storage, chain_name, msgs)?)
+    // Router-sent messages are assumed pre-verified and routable
+    // Otherwise, only route routable messages instantiated from CallContract
+    let msgs = if sender != router.address {
+        msgs.into_iter()
+            .unique()
+            .map(|msg| try_load_routable_msg(storage, msg))
+            .filter_map_ok(|msg| msg)
```

Comment:
I'm not sure if we should be ignoring unknown messages instead of failing and reporting the issue